install/installer/pkg/components/proxy/service.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package proxy
+
+import (
+	"fmt"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	"github.com/gitpod-io/gitpod/installer/pkg/config/v1/experimental"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func service(ctx *common.RenderContext) ([]runtime.Object, error) {
+	loadBalancerIP := ""
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		if cfg.WebApp != nil && cfg.WebApp.ProxyConfig != nil && cfg.WebApp.ProxyConfig.StaticIP != "" {
+			loadBalancerIP = cfg.WebApp.ProxyConfig.StaticIP
+		}
+		return nil
+	})
+
+	ports := map[string]common.ServicePort{
+		ContainerHTTPName: {
+			ContainerPort: ContainerHTTPPort,
+			ServicePort:   ContainerHTTPPort,
+		},
+		ContainerHTTPSName: {
+			ContainerPort: ContainerHTTPSPort,
+			ServicePort:   ContainerHTTPSPort,
+		},
+		MetricsContainerName: {
+			ContainerPort: PrometheusPort,
+			ServicePort:   PrometheusPort,
+		},
+	}
+	if ctx.Config.SSHGatewayHostKey != nil {
+		ports[ContainerSSHName] = common.ServicePort{
+			ContainerPort: ContainerSSHPort,
+			ServicePort:   ContainerSSHPort,
+		}
+	}
+
+	return common.GenerateService(Component, ports, func(service *corev1.Service) {
+		service.Spec.Type = corev1.ServiceTypeLoadBalancer
+		service.Spec.LoadBalancerIP = loadBalancerIP
+		service.Annotations["external-dns.alpha.kubernetes.io/hostname"] = fmt.Sprintf("%s,*.%s,*.ws.%s", ctx.Config.Domain, ctx.Config.Domain, ctx.Config.Domain)
```

Comment:
Thanks. We plan to change these annotations in a subsequent PR (they were just copied over rather than added in [0ec572a](https://github.com/gitpod-io/gitpod/pull/9764/commits/0ec572aa41f61ef8d1217ad90e3c454d7709563d) so we'll add tests for them there.